freeipa/src/main/java/com/sequenceiq/freeipa/repository/ChildEnvironmentRepository.java

```diff
@@ -15,9 +15,22 @@
 @Transactional(Transactional.TxType.REQUIRED)
 public interface ChildEnvironmentRepository extends CrudRepository<ChildEnvironment, Long> {
 
-    @Query("SELECT c.stack FROM ChildEnvironment c WHERE c.environmentCrn = :childEnvironmentCrn AND c.stack.accountId = :accountId")
+    @Query("SELECT c.stack FROM ChildEnvironment c WHERE c.environmentCrn = :childEnvironmentCrn AND c.stack.accountId = :accountId AND c.stack.terminated = -1")
     Optional<Stack> findParentStackByChildEnvironmentCrn(@Param("childEnvironmentCrn") String childEnvironmentCrn, @Param("accountId") String accountId);
 
+    @Query("SELECT c.stack FROM ChildEnvironment c LEFT JOIN FETCH c.stack.instanceGroups ig LEFT JOIN FETCH ig.instanceMetaData "
+            + "WHERE c.environmentCrn = :childEnvironmentCrn AND c.stack.accountId = :accountId")
+    List<Stack> findMultipleParentByEnvirnmentCrnWthListsEvenIfTerminated(
```

Comment:
The InstanceGroups are used in `StackToCreateFreeIpaRequestConverter::getInstanceGroupRequest()`

```
    private InstanceGroupRequest getInstanceGroupRequst(InstanceGroup instanceGroup) {
        InstanceGroupRequest request = new InstanceGroupRequest();
        request.setInstanceTemplateRequest(getInstanceTemplateRequest(instanceGroup.getTemplate()));
        request.setName(instanceGroup.getGroupName());
        request.setNetwork(getInststanceGroupNetworkRequest(instanceGroup.getInstanceGroupNetwork()));
        request.setNodeCount(instanceGroup.getNodeCount());
        request.setSecurityGroup(getSecurityGroupRequest(instanceGroup.getSecurityGroup()));
        request.setType(instanceGroup.getInstanceGroupType());
        return request;
    }
```

Although the instance metadata, is not used I think it makes sense to fully populate the instance groups so that future code only has 2 options: a full stack object with lists or a stack object without lists.

Reply:
I traced back, and this repository call is used only in `FreeIpaCustomCrnOrNameProvider`. Could you check too?

Reply:
Thanks, you are correct.